tests/driver_apa102/main.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the APA102 LED strip driver
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include "xtimer.h"
+#include "color.h"
+
+#include "apa102.h"
+#include "apa102_params.h"
+
+/**
+ * @brief   Switch to the next LED every 10ms
+ */
+#define STEP        (200 * US_PER_MS)
+
+/**
+ * @brief   Interval for dimming colors
+ */
+#define DIM         (100 * US_PER_MS)
+
+/**
+ * @brief   Step through brightness levels (0-255) in 32 steps
+ */
+#define BSTEP       (8U)
+
+/**
+ * @brief   Allocate the device descriptor
+ */
+static apa102_t dev;
+
+/**
+ * @brief   Allocate a color_rgb_t struct for each LED on the strip
+ */
+static color_rgba_t leds[APA102_PARAM_LED_NUMOF];
+
+static void setcolor(int color, uint8_t alpha)
+{
+    for (int i = 0; i < (int)APA102_PARAM_LED_NUMOF; i++) {
+        leds[i].alpha = alpha;
+        memset(&leds[i].color, 0, sizeof(color_rgb_t));
+        switch (color) {
+            case 0:
+                leds[i].color.r = 255;
+                break;
+            case 1:
+                leds[i].color.g = 255;
+                break;
+            case 2:
+                leds[i].color.b = 255;
+                break;
+        }
+    }
+}
+
+int main(void)
+{
+    int pos = 0;
+    int step = 1;
+    color_hsv_t col = { 0.0, 1.0, 1.0 };
+
+    puts("APA102 Test App");
+
+    /* initialize all LED color values to black (off) */
+    memset(leds, 0, sizeof(color_rgba_t) * APA102_PARAM_LED_NUMOF);
+
+    /* initialize the driver */
+    apa102_init(&dev, &apa102_params[0]);
+
+    puts("Initialization done.");
+
+    /* set to each red, green, and blue, and fade each color in and out */
+    for (int col = 0; col <= 2; col++) {
+        unsigned i = 0;
```

Comment:
unsigned causes trouble here. In the second loop `i`will wrap around and will afterwards count to "negative infinity"